torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -0,0 +1,221 @@
+import torch
+import itertools as it
+from typing import List, Optional, Dict
+from collections import namedtuple
+
+from torchaudio._torchaudio_decoder import (
+    _CriterionType,
+    _KenLM,
+    _LexiconDecoder,
+    _LexiconDecoderOptions,
+    _SmearingMode,
+    _Trie,
+    _Dictionary,
+    _create_word_dict,
+    _load_words,
+)
+
+
+__all__ = ["KenLMLexiconDecoder", "kenlm_lexicon_decoder"]
+
+
+Hypothesis = namedtuple("Hypothesis", ["tokens", "words", "score"])
+
+
+class KenLMLexiconDecoder:
+    def __init__(
+        self,
+        nbest: int,
+        lexicon: dict,
+        word_dict: _Dictionary,
+        tokens_dict: _Dictionary,
+        kenlm: _KenLM,
+        decoder_options: _LexiconDecoderOptions,
+        blank_token: str,
+        sil_token: str,
+        unk_word: str,
+    ) -> None:
+        """
+        KenLM CTC Decoder with Lexicon constraint.
+
+        Note:
+            To build the decoder, please use the factory function kenlm_lexicon_decoder.
+
+        Args:
+            nbest (int): number of best decodings to return
+            lexicon (dict): lexicon mapping of words to spellings
+            word_dict (_Dictionary): dictionary of words
+            tokens_dict (_Dictionary): dictionary of tokens
+            kenlm (_KenLM): n-gram KenLM language model
+            decoder_options (_LexiconDecoderOptions): parameters used for beam search decoding
+            blank_token (str): token corresopnding to blank
+            sil_token (str): token corresponding to silence
+            unk_word (str): word corresponding to unknown
+        """
+
```

Comment:
a few of these instance variables aren't referenced outside of the initializer, e.g. `self.decoder_options`, `self.lexicon`. can they be removed?